### PR TITLE
Set up database connection and create tables

### PR DESCRIPTION
**Pull request by rvl**
Relates to issue #143 and #154 

# Overview

- [x] Added partial `newDBLayer` function which sets up the db connection and runs migrations.
- [x] Added a small test case to check that the connection works and tables exist.
- [x] Add database queries for wallet metadata.

### CODE REVIEW

File: lib/core/cardano-wallet-core.cabal [hunk 1]
```diff
@@ -46,6 +46,7 @@ library
     , generic-lens
     , http-api-data
     , http-media
+    , lens
```
Comment (paweljakubas) [inline lib/core/cardano-wallet-core.cabal]: we use `generic-lens` already in codebase. If I remember well, it was decided to use this lens library. probably it is good idea in the last commit to comply with the rest of the code (if possible)

File: lib/core/cardano-wallet-core.cabal [hunk 2]
```diff
@@ -59,6 +60,13 @@ library
     , transformers
     , vector
 
+    , persistent
+    , persistent-sqlite
+    , conduit
+    , monad-logger
+    , mtl
```
Comment (paweljakubas) [inline lib/core/cardano-wallet-core.cabal]: we use `transformers` already. probably it is good idea in the last commit to comply with the rest of the code

**Comment by paweljakubas** [comment]
@rvl Please notice that the branch was rebased against `rvl/147/sql-schema` (which was rebased before against master) and forced pushed, so if #227 is merged then you can put it against master and it should compile and be green (made needed fixes and also cleaned imports). I started addressing `PrivateKey` table and stumbled into problem how to really represent  `(Key 'RootK XPrv, W.Hash "encryption")` . I tried to represent them as `(Text, Text)` via ByteString but then have problem to return from db to `Key 'RootK XPrv`

**Comment by rvl** [comment]
@paweljakubas Thanks :smile:

File: lib/core/cardano-wallet-core.cabal (hunk 2 above)
Comment (paweljakubas) [reply lib/core/cardano-wallet-core.cabal]: done

**Comment by paweljakubas** [comment]
@rvl Please read it before you continue working : 
1. `rvl/147/sql-conn` is rebased against master, compiles and should be green
2. I ported all tests from MVarSpec adding SQL flavor (especially to wallet state which is represented as `(UTxO, PendingTx)`)
3. A number of tests does not pass - I denoted them as `xit` rather than `it` (at this moment 16 tests do not pass)
4. I moved shared parts between MVar and SQL to DBSpec - still properties can also be moved to DBspec and shared as they are almost the same - maybe we can do it at the end together with lens refactoring

File: lib/core/cardano-wallet-core.cabal (hunk 1 above)
Comment (paweljakubas) [reply lib/core/cardano-wallet-core.cabal]: done

**Comment by paweljakubas** [comment]
@rvl Two things done:
1. `ErrWalletAlreadyExists` supported
2. Handled quite rare :  
```
test/unit/Cardano/Wallet/DBSpec.hs:403:9:
 1) Cardano.Wallet.DB.Sqlite, put . read yields a result, Wallet Metadata
      Falsifiable (after 30 tests):
        (PrimaryKey (WalletId {getWalletId = 1d05f4517d2decf9b118b7a48e46a534b42f67a4}),WalletMetadata {name = WalletName {getWalletName = "bulbazaur"}, passphraseInfo = Just (WalletPassphraseInfo {lastUpdatedAt = 1864-04-28 23:32:45.015596992627 UTC}), status = Restoring (Quantity (Percentage {getPercentage = 100})), delegation = NotDelegating})
      expected: Just (WalletMetadata {name = WalletName {getWalletName = "bulbazaur"}, passphraseInfo = Just (WalletPassphraseInfo {lastUpdatedAt = 1864-04-28 23:32:45.015596992627 UTC}), status = Restoring (Quantity (Percentage {getPercentage = 100})), delegation = NotDelegating})
       but got: Just (WalletMetadata {name = WalletName {getWalletName = "bulbazaur"}, passphraseInfo = Just (WalletPassphraseInfo {lastUpdatedAt = 1864-04-28 23:32:45.015596992627 UTC}), status = Ready, delegation = NotDelegating})
```
Added unit tests exemplifying this case plus changed `Gen` .

File: lib/core/src/Cardano/Wallet/DB.hs [hunk 3]
```diff
@@ -35,7 +39,7 @@ import Data.Map.Strict
 -- | A Database interface for storing various things in a DB. In practice,
 -- we'll need some extra contraints on the wallet state that allows us to
 -- serialize and unserialize it (e.g. @forall s. (Serialize s) => ...@)
-data DBLayer m s t = DBLayer
+data (IsOurs s, NFData s, Show s, TxId t) => DBLayer m s t = DBLayer
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB.hs]: `Show s` & `IsOurs s` :thinking: ?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 4]
```diff
@@ -46,7 +128,7 @@ share
 Wallet
     walTableId                 W.WalletId     sql=wallet_id
     walTableName               Text           sql=name
-    walTablePassphraseLastUpdatedAt  UTCTime  sql=passphrase_last_updated_at
+    walTablePassphraseLastUpdatedAt  UTCTime Maybe  sql=passphrase_last_updated_at
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs:105]: PTSD triggered :grimacing: ...

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 5]
```diff
@@ -1,43 +1,125 @@
...
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import qualified Data.Text as T
+import qualified Database.Sqlite as Sqlite
 
 share
-    [ mkPersist sqlSettings { mpsPrefixFields = False }
+    [ mkPersist sqlSettings'
+    , mkDeleteCascade sqlSettings'
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs:96]: :pray: Hell yeah. Finally.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 6]
```diff
@@ -94,7 +177,7 @@ TxIn
     txInputTableSourceIndex  Word32      sql=source_index
 
     Primary txInputTableTxId txInputTableSourceTxId txInputTableSourceIndex
-    Foreign TxMeta fk_tx_meta_tx_in txInputTableTxId txInputTableWalletId
+    Foreign TxMeta fk_tx_meta_tx_in txInputTableTxId txInputTableWalletId -- fixme: remove
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: Why `remove` ? Could the comment give a bit more details about what's wrong, and `how` this should or could be fixed ?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 7]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+unsafeRunQuery
+    :: SqlBackend
+    -> SqlPersistM a
+    -> IO a
+unsafeRunQuery conn = runResourceT . runNoLoggingT . flip runSqlConn conn
+
+runQuery
+    :: SqlBackend
+    -> SqlPersistM a
+    -> ExceptT Sqlite.SqliteException IO a
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: Question: do we know what happens in this `IO` here? Is there any way to lower the constraint down at this level? My concern here is that `runQuery` could be throwing in the wild and bypass the actual `ExceptT` :thinking: ? 

What if the connection is gone, do we get an `SqliteException`, or does it throw? etc .. Is it worth adding some `catch` or `try` and move some exceptions at the value level ?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 8]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+-- | Sets up a connection to the SQLite database.
+--
+-- Database migrations are run to create tables if necessary.
+--
+-- If the given file path does not exist, it will be created by the sqlite
+-- library.
+newDBLayer
+    :: forall s t. (IsOurs s, PersistState s, NFData s, Show s, W.TxId t)
+    => Maybe FilePath
+       -- ^ Database file location, or Nothing for in-memory database
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs:259]: What about a type isomorphic to `Maybe FilePath` that would convey this idea:

```hs
data Storage = InMemory | InFile FilePath
```

?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 9]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+            map (PrimaryKey . unWalletKey) <$> selectKeysList [] []
+
+        {-----------------------------------------------------------------------
+                                    Checkpoints
+        -----------------------------------------------------------------------}
+
+        , putCheckpoint = \(PrimaryKey wid) cp ->
+            ExceptT $ unsafeRunQuery conn $ Right <$> do
+                deleteCheckpoints wid (Proxy :: Proxy s) -- clear out all checkpoints
+                deleteLooseTransactions wid -- clear transactions
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: This is in case we would have some data inconsistency in the database? Is it worth checking the `count` of those queries and, output a warning if anything is beyond `0`?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 10]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+                    s <- selectState (checkpointId cp)
+                    pure (checkpointFromEntity cp utxo ins outs <$> s)
+                Nothing -> pure Nothing
+
+        {-----------------------------------------------------------------------
+                                   Wallet Metadata
+        -----------------------------------------------------------------------}
+
+        , putWalletMeta = \(PrimaryKey wid) meta ->
+            ExceptT $ unsafeRunQuery conn $
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: Why are all the queries running as `unsafeRunQuery` :/ ?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 11]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+        {-----------------------------------------------------------------------
+                                       Lock
+        -----------------------------------------------------------------------}
+
+        , withLock = \_action -> error "withLock to be implemented"
+
+        }
+
+----------------------------------------------------------------------------
+-- Conversion between Persistent table types and wallet types
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs:335]: Seeing conversions below, what about prefixing all db types by `DB` or `SQL`, like `SQLWallet` `SQLPrivateKey` etc ... So that it's obvious where types belong ... (As we do for API types).

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 12]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+        , txOutputTableWalletId = wid
+        , txOutputTableIndex = ix
+        , txOutputTableAddress = W.address txOut
+        , txOutputTableAmount = W.coin txOut
+        }
+    utxo = [ UTxO wid sl (TxId input) ix addr coin
+           | (W.TxIn input ix, W.TxOut addr coin) <- utxoMap ]
+    utxoMap = Map.assocs (W.getUTxO (W.totalUTxO wal))
+
+dist :: (a -> b -> c) -> (a, [b]) -> [c]
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: For small utility functions like that, I find it very useful to put a small doc comment containing an example. It removes some unnecessary brain-load when reading it:

```
-- | Distribute `a` accross many `b`s using the given function.

>>> dist TxOut (addr, [Coin 1, Coin 42, Coin 14])
[TxOut addr (Coin 1), TxOut addr (Coin 42), TxOut addr (Coin 14)]
```

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 13]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+checkpointFromEntity
+    :: forall s t. (IsOurs s, NFData s, Show s, W.TxId t)
+    => Checkpoint
+    -> [UTxO]
+    -> [TxIn]
+    -> [TxOut]
+    -> s
+    -> W.Wallet s t
+checkpointFromEntity (Checkpoint _ tip) utxo ins outs s =
+    W.Wallet utxo' pending tip s
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: I'd be keen to call this `UnsafeWallet`, or put the constructor in a separate `.Internal` module and add a rule in hlint to make sure that this is only used in this particular file. Exposing the constructor will lead to abuse and since most of our guarantees regarding the wallet state relies on the fact that only "applyBlock" is used to modify the state of the wallet, it is rather important to enforce we preserve that.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 14]
```diff
@@ -1,43 +1,125 @@
...
+import Numeric.Natural
+    ( Natural )
+import System.IO
+    ( stderr )
+import System.Log.FastLogger
+    ( fromLogStr )
 
+import qualified Cardano.Wallet.Primitive.AddressDiscovery as W
+import qualified Cardano.Wallet.Primitive.Model as W
 import qualified Cardano.Wallet.Primitive.Types as W
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs:89]: The `W.` across the file feels very noisy to me :disappointed: ... I am not sure about this style.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 15]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+----------------------------------------------------------------------------
+
+checkpointId :: Checkpoint -> (W.WalletId, W.SlotId)
+checkpointId cp = (checkpointTableWalletId cp, checkpointTableSlot cp)
+
+class PersistState s where
+    insertState :: MonadIO m => (W.WalletId, W.SlotId) -> s -> ReaderT SqlBackend m ()
+    selectState :: MonadIO m => (W.WalletId, W.SlotId) -> ReaderT SqlBackend m (Maybe s)
+    deleteState :: MonadIO m => Proxy s -> W.WalletId -> ReaderT SqlBackend m ()
+    addressScheme :: Proxy s -> AddressScheme
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: I am not sure to understand the rationale behind `insertState` and `selectState` ? What are their use? Can we add comments please :pray:

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 16]
```diff
@@ -105,49 +188,589 @@ TxOut
...
+    { txMetaTableTxId = TxId txid
+    , txMetaTableWalletId = wid
+    , txMetaTableStatus = meta ^. #status
+    , txMetaTableDirection = meta ^. #direction
+    , txMetaTableSlotId = meta ^. #slotId
+    , txMetaTableAmount = getAmount (meta ^. #amount)
+    }
+    where getAmount (Quantity n) = n
+
+-- note: TxOut records must already be sorted by index
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: Sounds like a good property-test to have, or an invariant in the code. Either way, the note isn't sufficient here.

**Review by KtorZ — COMMENTED** [review]
Looks good overall. Although the `Sqlite` file lacks of documentation IMO. Don't forget that we are the first consumers of that code and that documentation. Be verbose on details, give concrete examples, make it easy to consume the module :pray:

File: lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs [hunk 17]
```diff
@@ -42,9 +42,10 @@ module Cardano.Wallet.Primitive.AddressDiscovery
     , accountPubKey
     , mkAddressPool
     , lookupAddress
+    , indexedAddresses -- fixme: internals exposed for sql
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs]: NOTE: `addresses` is already exposed and does exactly what the SQL engine needs I believe.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 18]
```diff
@@ -105,49 +153,221 @@ TxOut
...
+    return $ DBLayer
+
+        {-----------------------------------------------------------------------
+                                      Wallets
+        -----------------------------------------------------------------------}
+
+        { createWallet = \(PrimaryKey wid) _cp meta ->
+            ExceptT $ unsafeRunQuery conn $
+            selectWallet wid >>= \case
+                Just _ ->
```
Comment (akegalj) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: what would happen if we didn't make this check? Would `insert_` fail?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 19]
```diff
@@ -105,49 +153,221 @@ TxOut
...
+            Just (W.WalletPassphraseInfo passInfo) -> Just passInfo
+    , WalTableStatus =. meta ^. #status
+    , WalTableDelegation =. delegationToText (meta ^. #delegation)
+    ]
+
+metadataFromEntity :: Wallet -> W.WalletMetadata
+metadataFromEntity wal = W.WalletMetadata
+    { name = W.WalletName (walTableName wal)
+    , passphraseInfo = case walTablePassphraseLastUpdatedAt wal of
+            Just time -> Just $ W.WalletPassphraseInfo time
```
Comment (akegalj) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: why not 
```
passphraseInfo = W.WalletPassphraseInfo <$> walTablePassphraseLastUpdatedAt wal
```

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 20]
```diff
@@ -105,49 +153,221 @@ TxOut
...
+    , walTableDelegation = delegationToText $ meta ^. #delegation
+    , walTableAddressScheme = Sequential
+    }
+
+mkWalletMetadataUpdate :: W.WalletMetadata -> [Update Wallet]
+mkWalletMetadataUpdate meta =
+    [ WalTableName =. meta ^. #name . coerce
+    , WalTablePassphraseLastUpdatedAt =. case meta ^. #passphraseInfo of
+            Nothing -> Nothing
+            Just (W.WalletPassphraseInfo passInfo) -> Just passInfo
```
Comment (akegalj) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: why not:
```
WalTablePassphraseLastUpdatedAt =.  W.lastUpdateAt <$> meta ^. #passphraseInfo

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 21]
```diff
@@ -105,49 +153,221 @@ TxOut
...
+delegationFromText Nothing = W.NotDelegating
+delegationFromText (Just pool) = W.Delegating (W.PoolId pool)
+
+mkWalletEntity :: W.WalletId -> W.WalletMetadata -> Wallet
+mkWalletEntity wid meta = Wallet
+    { walTableId = wid
+    , walTableName = meta ^. #name . coerce
+    , walTablePassphraseLastUpdatedAt = case meta ^. #passphraseInfo of
+            Nothing -> Nothing
+            Just (W.WalletPassphraseInfo passInfo) -> Just passInfo
```
Comment (akegalj) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: https://github.com/input-output-hk/cardano-wallet/pull/247/files#r284836455

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 18 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: if we do not check this then we cannot handle `ErrWalletAlreadyExists` - there are tests that are checking this and my believe is that we do not want to do it

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 19 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: Yep, fixed.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 18 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: There would be an exception from SQLite. Actually I think it will be better to handle this exception than do select + insert. Fixed.

File: lib/core/src/Cardano/Wallet/DB.hs (hunk 3 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB.hs]: I have cut it.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 4 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs:105]: I don't understand what you mean. Should something be changed?

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 12 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: OK this code has been cut for now but I will address the request for documentation in the next PR.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 13 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: This code has been deleted and will reappear in a later PR.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 8 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs:259]: I don't think it's necessary.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 15 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: This code has been cut and will appear in a second PR with comments.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 16 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: This code has been cut and will reappear in a subsequent PR.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 14 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs:89]: It is very reasonable to have qualified name prefixing in a module where both SQL table types and application types are present.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 9 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: This code has been cut and will reappear in a later PR.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 7 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: I removed this function, but this is a good point.

Any exception that we "expect" (e.g. db constraint), or can recover from, should be put into the DBLayer return types.

For all other exceptions (e.g. connection gone - not sure whether that's even possible in sqlite) then it's fine to exit the wallet with an error message.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 10 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: I made the function safe by renaming it.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 11 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs:335]: It will be shorter and neater without name prefixes. All these table and column types are local to the Sqlite module. The actual wallet types are qualified with W.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 10 above)
Comment (KtorZ) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: > I made the function safe by renaming it.

Hmmm ? :thinking:

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 4 above)
Comment (KtorZ) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs:105]: Just the alignment that is off because of the long name. It good for the eyes, but nothing really important here.

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 14 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs:89]: in previous codebase we had `Kernel` prefix

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 22]
```diff
@@ -103,51 +158,219 @@ TxIn
...
+-- Database migrations are run to create tables if necessary.
+--
+-- If the given file path does not exist, it will be created by the sqlite
+-- library.
+newDBLayer
+    :: Maybe FilePath
+       -- ^ Database file location, or Nothing for in-memory database
+    -> IO (DBLayer IO s t)
+newDBLayer fp = do
+    conn <- createSqliteBackend fp logStderr
```
Comment (paweljakubas) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: wil this `logStderr` enable all logs (CREATE_TABLE, INSERT, ...) as a default or errors only? I would give my votes for the second option

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs [hunk 23]
```diff
@@ -103,51 +158,219 @@ TxIn
...
+
+mkWalletEntity :: W.WalletId -> W.WalletMetadata -> Wallet
+mkWalletEntity wid meta = Wallet
+    { walTableId = wid
+    , walTableName = meta ^. #name . coerce
+    , walTablePassphraseLastUpdatedAt =
+        W.lastUpdatedAt <$> meta ^. #passphraseInfo
+    , walTableStatus = meta ^. #status
+    , walTableDelegation = delegationToText $ meta ^. #delegation
+    , walTableAddressScheme = Sequential
```
Comment (paweljakubas) [inline lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: This probably has to be addressed in another PR. Either removed or detected whether we are Random or Sequential

**Review by paweljakubas — APPROVED** [review]
LGTM

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 23 above)
Comment (rvl) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: Yes there is not yet reading/writing of the wallet state.
However we may even remove this field because the address scheme is fixed by a type parameter of DBLayer.

Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: exactly. This is in line what @KtorZ wrote on slack yesterday:
```
Then, we'll have one

- external seq wallet layer + seq db layer
- fully owned  seq wallet + seq db layer
- fully owned rnd wallet + rnd db layer
```
Address scheme is set for each db instance. Hence, `walTableAddressScheme` is probably superflouous

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 22 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: indeed, it introduces logs like : 
```Cardano.Wallet.DB.Sqlite
  Wallet table
Migrating: CREATE TABLE "wallet"("wallet_id" VARCHAR NOT NULL,"name" VARCHAR NOT NULL,"passphrase_last_updated_at" TIMESTAMP NULL,"status" INTEGER NOT NULL,"delegation" VARCHAR NULL, PRIMARY KEY ("wallet_id"))
CREATE TABLE "wallet"("wallet_id" VARCHAR NOT NULL,"name" VARCHAR NOT NULL,"passphrase_last_updated_at" TIMESTAMP NULL,"status" INTEGER NOT NULL,"delegation" VARCHAR NULL, PRIMARY KEY ("wallet_id")); []
```

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 23 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: removed address scheme

File: lib/core/src/Cardano/Wallet/DB/Sqlite.hs (hunk 22 above)
Comment (paweljakubas) [reply lib/core/src/Cardano/Wallet/DB/Sqlite.hs]: added more customizable logging function (one can choose LogLevel)
